modules/end-user-guide/pages/git-credential-store.adoc

```diff
@@ -1,25 +1,79 @@
+** For CLI users: `{orch-cli}` is installed in the operating system you are using. See link:https://docs.openshift.com/container-platform/4.10/cli_reference/openshift_cli/getting-started-cli.html#installing-openshift-cli[Installing the OpenShift CLI].
+endif::[]
+
+* For CLI users: link:https://www.gnu.org/software/coreutils/base64[`base64`] is installed in the operating system you are using.
 
 .Procedure
-. Prepare Git credential file in the link:https://git-scm.com/docs/git-credential-store#_storage_format[Storage format].
-. Encode content of the file to the base64 format.
-. Create a new {platforms-name} secret in the {platforms-name} {orch-namespace} where a {prod-short} workspace will be created.
+. In your home directory, locate and open your `.git-credentials` file if you already have it. Alternatively, if you do not have this file, then save a new `.git-credentials` file, using the following link:https://git-scm.com/docs/git-credential-store#_storage_format[Git credentials storage format]. Each credential is stored on its own line in the file:
+
++
+[subs="+quotes,+attributes,+macros"]
+----
+https://__<username>__:__<token>__@__<git_server_hostname>__
+----
+
++
+.A line in a `.git-credentials` file
+====
+`https://trailblazer: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@github.com`
+====
 
-* The labels of the secret that is about to be created must match the set of labels configured in `che.workspace.provision.secret.labels` property of {prod-short}. The default labels are:
+. Select credentials from your `.git-credentials` file to provide to the Secret. Encode the selected credentials to Base64 for the next step.
 
-* `app.kubernetes.io/part-of: che.eclipse.org` 
-* `app.kubernetes.io/component: workspace-secret`:
+. Create a new {platforms-name} secret in the user's {orch-namespace}.
 +
-include::example$snip_{project-context}-git-credential-file-as-kubernetes-secret-into-the-file.adoc[leveloffset=+1]
+[source,yaml,subs="+quotes,+attributes,+macros"]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: git-credentials-secret
+  labels:
+    controller.devfile.io/git-credential: 'true' <1>
+    controller.devfile.io/watch-secret: 'true'
+data:
+  credentials: __<Base64_content_of_.git-credentials>__ <2>
+----
++
+<1> The `controller.devfile.io/git-credential` label marks the secret as containing Git credentials. All Git credential secrets in the same {orch-namespace} will be merged into a single secret that will be mounted to the {devworkspace} containers. All the original secrets are kept intact.
+<2> The Git credentials store content encoded with `base64`.
+
++
+[NOTE]
+====
+Optionally, a container path to mount the merged Git credentials Secret can be provided. This can be done by setting the `controller.devfile.io/mount-path` annotation to a Git credntial Secret. For example, if set to `/tmp/git-credentials`, the merged Git credentials store is mounted at `/tmp/git-credentials/credentials`.
```

Comment:
@dkwon17,

1. Does the `controller.devfile.io/mount-path:` line enable the user to specify both the file path and file name?

2. Imagine, if you test it, is there a file path and file name this Secret defaults to when applied according to this procedure? In other words, will there appear a file in some directory if you apply such Secret without the `controller.devfile.io/mount-path:` line? If so, then it's good to explicitly mention where and as which file the user can expect to find it.

3. Is this secret targeting specific containers, workspace containers? If so, would it be helpful to users to understand the concepts if we mention the target category of containers here?

4. I can't predict whether the following will be relevant to some users, but the page that we're linking to here on line 26 also informs us that
![image](https://user-images.githubusercontent.com/78564304/163136554-5b687d05-4159-4eba-85b5-6f06d4af7cc0.png)
So I wonder whether it would make sense to use this file path and file name in the example, if such example is different from the default.

```suggestion
You can optionally specify a custom path where to mount the merged Git credentials Secret as a file:
[source,yaml,subs="+quotes,+attributes,+macros"]
----
metadata:
  ...
  annotations:
    controller.devfile.io/mount-path: __<absolute_path_inside_containers>__
----
For example: `controller.devfile.io/mount-path: ~/.git-credentials`   
```

Reply:
> is there a file path that this Secret defaults to when applied according to this procedure?

Yes, it defaults to "/": https://github.com/devfile/devworkspace-operator/blob/41e8ccb8623b80f0829c605f372e468bdf9fdda0/pkg/provision/automount/templates.go#L148

Per our discussion, I've re-added the mount-path in the example yaml:
<img width="546" alt="image" src="https://user-images.githubusercontent.com/83611742/163251712-a7f2ad3d-1f67-4679-a101-b0e12d723e08.png">